app/controllers/advertisements_controller.rb

```diff
@@ -0,0 +1,52 @@
+class AdvertisementsController < ApplicationController
+  before_action :set_advertisement, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @advertisements = Advertisement.all
+  end
+
+  def show
+  end
+
+  def new
+    @advertisement = Advertisement.new
+  end
+
+  def edit
+  end
+
+  def create
+    @advertisement = Advertisement.new(advertisement_params)
+
+    if @advertisement.save
+      redirect_to @advertisement, notice: 'Advertisement was successfully created.'
```

Comment:
Line is too long. [83/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.